Regression in oas3-examples-value-or-externalValue: examples in examples is asked to be OpenAPI compliant

**Describe the bug**

Examples properties in OpenAPI Examples are reported as invalid although they are the actual examples and contain no value or external value, of course. This seems to be a regression (or just something related) of my fix in #899.

**To Reproduce**

Validate https://raw.githubusercontent.com/Open-EO/openeo-api/draft/openapi.yaml

Response (reports for oas3-unused-components-schema may be valid, reports for oas3-examples-value-or-externalValue are likely invalid):

```
> spectral lint openapi.yaml

OpenAPI 3.x detected

c:/dev/openeo-api/openapi.yaml
 1592:25  warning  oas3-examples-value-or-externalValue  Examples should have either a `value` or `externalValue` field.
 1596:25  warning  oas3-examples-value-or-externalValue  Examples should have either a `value` or `externalValue` field.
 1600:25  warning  oas3-examples-value-or-externalValue  Examples should have either a `value` or `externalValue` field.
 3330:30  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 3398:16  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 4264:21  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 4281:21  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 4347:34  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 4353:32  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 4364:24  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 4680:17  warning  oas3-examples-value-or-externalValue  Examples should have either a `value` or `externalValue` field.
 4734:11  warning  oas3-examples-value-or-externalValue  Examples should have either a `value` or `externalValue` field.
 5414:18  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 5453:23  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 5465:20  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 5479:23  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 5491:28  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 5505:25  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.
 5521:31  warning  oas3-unused-components-schema         Potentially unused components schema has been detected.

✖ 19 problems (0 errors, 19 warnings, 0 infos, 0 hints)
```

**Expected behavior**

No warnings for rule `oas3-examples-value-or-externalValue`.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment (remove any that are not applicable):**
 - Library version: 5.7.1
 - OS: Win10

--- a/src/rulesets/oas/__tests__/oas3-examples-value-or-externalValue.ts
+++ b/src/rulesets/oas/__tests__/oas3-examples-value-or-externalValue.ts
@@ -67,6 +67,26 @@ describe('oas3-examples-value-or-externalValue', () => {
     expect(results.length).toEqual(0);
   });
 
+  test('will not validate example property placed in schema under components schema', async () => {
+    const results = await s.run({
+      components: {
+        schemas: {
+          pet: {
+            properties: {
+              examples: {
+                type: 'array',
+              },
+              example: {
+                type: 'integer',
+              },
+            },
+          },
+        },
+      },
+    });
+    expect(results.length).toEqual(0);
+  });
+
   test('multiple examples - validate all value or externalValue', async () => {
     const results = await s.run({
       components: {

--- a/src/rulesets/oas/__tests__/oas3-examples-value-or-externalValue.ts
+++ b/src/rulesets/oas/__tests__/oas3-examples-value-or-externalValue.ts
@@ -67,7 +67,7 @@ describe('oas3-examples-value-or-externalValue', () => {
     expect(results.length).toEqual(0);
   });
 
-  test('will not validate example property placed in schema under components schema', async () => {
+  test('will not validate properties in schemas that are literally named example or examples', async () => {
     const results = await s.run({
       components: {
         schemas: {

--- a/src/rulesets/oas/index.json
+++ b/src/rulesets/oas/index.json
@@ -518,7 +518,6 @@
       "given": [
         "$.components.examples[*]",
         "$.paths[*][*]..content[*].examples[*]",
-        "$.components.content[*].examples[*]",
         "$.paths[*][*]..parameters[*].examples[*]",
         "$.components.parameters[*].examples[*]",
         "$.paths[*][*]..headers[*].examples[*]",

--- a/src/rulesets/oas/index.json
+++ b/src/rulesets/oas/index.json
@@ -515,7 +515,15 @@
       "recommended": true,
       "formats": ["oas3"],
       "type": "style",
-      "given": ["$.paths..examples[*]", "$.components..examples[*]"],
+      "given": [
+        "$.components.examples[*]",
+        "$.paths[*][*]..content[*].examples[*]",
+        "$.components.content[*].examples[*]",
+        "$.paths[*][*]..parameters[*].examples[*]",
+        "$.components.parameters[*].examples[*]",
+        "$.paths[*][*]..headers[*].examples[*]",
+        "$.components.headers[*].examples[*]"
+      ],
       "then": {
         "function": "xor",
         "functionOptions": {

